calico/openstack/mech_calico.py

```diff
@@ -675,6 +683,52 @@ def resync_endpoints(self, context):
                 )
                 self.transport.endpoint_created(port)
 
+        # Finally, scan each of the ports in changes_ports. Work out if there
+        # are any differences. If there are, write out to etcd.
+        changed_endpoints = (e for e in endpoints if e.id in changes_ports)
+
+        for endpoint in changed_endpoints:
+            # Get the endpoint data from etcd.
+            try:
+                endpoint = self.transport.get_endpoint_data(endpoint)
+            except etcd.EtcdKeyNotFound:
+                # The endpoint is gone. That's fine.
+                LOG.info("Failed to update deleted endpoint %s", endpoint.id)
+                continue
+
+            with context.session.begin(subtransactions=True):
+                port = self.db.get_port(context, endpoint.id)
+
+            # Get the data for both.
+            try:
+                etcd_data = json.loads(endpoint.data)
+            except Exception:
+                # If the JSON data is bad, just ignore it. We can't blow up
+                # here because that will trigger a new resync on another node
+                # that will blow *it* up as well. Just tolerate it.
+                LOG.exception("Bad JSON data in key %s", endpoint.key)
+                continue
```

Comment:
Yeah, we should. We do it in the refactor, weirdly, so I'm inclined to leave it in place for now and add it in the refactor code.